kubernetes-services/work-scheduler/app/util/k8s.ts

```diff
@@ -0,0 +1,21 @@
+import { k8sApi } from '../workers/scheduler';
+
+/**
+ * Get the number of pods running for a service
+ * @param serviceId - The service ID for which to get the number of pods
+ * @param namespace - The namespace in which to look for pods
+ * @returns The number of pods running for the service
+ * @throws An error if the Kubernetes API call fails
+ **/
+export async function getPodsCountForService(serviceId: string, namespace = 'harmony'): Promise<number> {
+  // Get all pods in the namespace
+  const allPodsResponse = await k8sApi.listNamespacedPod(namespace);
```

Comment:
I'm curious how often this is called and what the performance of this function is - I think I'll want to look at that while testing - what do you think of adding a debug log line that logs the duration?

Reply:
Added timing/logging